src/main/java/org/broadinstitute/hellbender/utils/test/VariantContextTestUtils.java

```diff
@@ -70,10 +79,139 @@ static Object normalizeScientificNotation(final Object attribute){
         return attribute;
     }
 
+    /**
+     * Method which reorders the fields of a Variant Context so that its AltAlleles are in alphabetical order.
+     *
+     * NOTE: this method relies on the correctness of AlleleSubsettingUtils.subsetAlleles() for reordering alleles
+     *
+     * @param vc        Variant context to reorder
+     * @param header
+     * @return
+     */
+    public static VariantContext sortAlleles(final VariantContext vc, final VCFHeader header){
+        final List<Allele> originalAltAlleles = vc.getAlternateAlleles();
+        final List<Allele> sortedAltAlleles = originalAltAlleles.stream().sorted().collect(Collectors.toList());
+        final List<Allele> sortedAlleles = new ArrayList<>(vc.getNAlleles());
+
+        sortedAlleles.add(vc.getReference());
+        sortedAlleles.addAll(sortedAltAlleles);
+
+        final VariantContextBuilder result = new VariantContextBuilder(vc);
+        result.alleles(sortedAlleles);
+
+        GenotypesContext newGT = AlleleSubsettingUtils.subsetAlleles(vc.getGenotypes(),2,vc.getAlleles(),sortedAlleles,
+                GenotypeAssignmentMethod.SET_TO_NO_CALL, vc.getAttributeAsInt(VCFConstants.DEPTH_KEY,0));
+
+        final HashMap<String, Object> newAttributes = new HashMap<>(vc.getAttributes());
+        newAttributes.replaceAll((key, value) -> updateAttribute(key, value, vc.getAlleles(), sortedAlleles,
+                                                                 header.hasInfoLine(key)?header.getInfoHeaderLine(key).getCountType():VCFHeaderLineCount.UNBOUNDED, vc.getGenotypes().getMaxPloidy(2)));
+
+        // The above will have built new genotype for PL,AD, and SAC fields excluding the GT and GQ field, thus we must re-add them for comparison.
+        for (int i = 0; i < newGT.size(); i++) {
+            newGT.replace(new GenotypeBuilder(newGT.get(i)).GQ(vc.getGenotype(i).getGQ()).phased(vc.getGenotype(i).isPhased()).alleles(vc.getGenotype(i).getAlleles()).make());
+        }
+
+        result.attributes(newAttributes).genotypes(newGT);
+
+        return result.make();
+
+    }
+
+    private static Object updateAttribute(final String key, final Object value,
+                                          final List<Allele> originalAlleles, final List<Allele> sortedAlleles,
+                                          final VCFHeaderLineCount count, int ploidy) {
+        if (key.startsWith("AS_")) {
```

Comment:
@lbergelson We probably well should, the issue is the fact that unless we are holding on to a `VariantAnnoationEngine` object at the time of asking that question there is no map of key ->  annotation to poll unless we were to hold onto that information in someplace like `GATKVCFConstants`. This could probably be gotten around by having there be a static implementation of AnnoatorEngine kicking around which has instantiated all of the annotations though.